Initial checkin

michaelfigura opened the pull request.

michaelfigura committed:
Initial checkin

Frank-Wiebeler commented on CottonHarvesterHIDFileImportPlugin/Plugin.cs:
```diff
@@ -0,0 +1,141 @@
+        string IPlugin.Owner => "AgGateway";
+        public IList<IError> Errors => throw new NotImplementedException();
+
+        public IList<ApplicationDataModel> RunPlugin()
```
As this is a test functionality, please place it outside the Plugin.

Frank-Wiebeler commented on CottonHarvesterHIDFileImportPlugin/Plugin.cs:
```diff
@@ -0,0 +1,141 @@
+            //A plugin publisher can choose to create one or multiple application data models as appropriate for the data
+            IList<ApplicationDataModel> admList = new List<ApplicationDataModel>();
+            ApplicationDataModel adm = new ApplicationDataModel();            
+            string dataPath = @"C:\Projects\OAGi\ADAPT\JD";
```
(When placed outside the plugin) Can this path be made relative?

Frank-Wiebeler commented on CottonHarvesterHIDFileImportPlugin/Plugin.cs:
```diff
@@ -0,0 +1,141 @@
+                ApplicationDataModel adm = new ApplicationDataModel();
+                adm.Catalog = new Catalog() { Description = $"ADAPT data transformation of Publisher data {DateTime.Now.ToShortDateString()} {dataPath}" };
+                //var loads = new List<AgGateway.ADAPT.ApplicationDataModel.LoggedData.Load>();
+                //models.Add(adm);
```
Is there a reason why the commented lines shall be part of the PR?

Frank-Wiebeler commented on CottonHarvesterHIDFileImportPlugin/PublisherDataModel/ExampleData/ExampleDataSet.cs:
```diff
@@ -0,0 +1,63 @@
+
+namespace CottonHarvesterHIDFileImportPlugin.PublisherDataModel.ExampleData
+{
+    /// <summary>
```
Is this file required anyway?

Frank-Wiebeler commented on CottonHarvesterHIDFileImportPlugin/PublisherDataModel/Model/HIDData.cs:
```diff
@@ -0,0 +1,246 @@
+                    continue;
+                }
+
+                hidRecord.ModuleID = currentRecord[0];
```
Number of entries in currentRecord should be checked to avoid data corruption

Frank-Wiebeler reviewed (changes requested):
Thank you for this new plugin,
I just have a few minor change requests:

- Rename the .myjson-Files to .json
- It might make sense to move the ExampleData folder outside the PublisherDataModel to avoid mixing up code and TestData

michaelfigura committed:
1. I updated the Example Data Importer test harness with comments describing how to use it. I also changed the File Name property to a wild card “*”. Finally, I deleted the Example Data folder. Any future developer must provide their own test input file.

michaelfigura committed:
2. Changed ConvertFlatFileToCustomModel method name to ConvertFlatFileToADAPTFarmModel. Got rid of the JSON converter method, since it’s unused.

michaelfigura committed:
4. Updated all HID values to be “” just in case the length of the variable equals zero.

michaelfigura committed:
Got rid of the use of OMSourceId.

michaelfigura committed:
Updated OMCode, but had to comment it out.

michaelfigura committed:
1. I updated the Example Data Importer test harness with comments describing how to use it.
2. I changed the File Name property to a wild card “*”.
3. I deleted the Example Data folder. Any future developer must provide their own test input file.
4. Changed ConvertFlatFileToModel method name to ConvertFlatFileToJDHIDModel. Got rid of the JSON converter method, since it’s unused.
5. I removed all underscores from variable names in DataMapper.cs
6. Changed SearchOpteion.AllDirectories to SearchOption.TopDirectoryOnly
7. Researched File.ReadAllLines, and it returns a string[], which is the same. Researched File.ReadLines, and it returns an Enumberable which immediately requires a String.Join(Environment.NewLine, lines), which is also the same. The way I’m performing this functionality should be fine.
8. Updated all HID values to be “” just in case the length of the variable equals zero.
9. Got rid of the use of OMSourceId.
10. Updated all Obs code to use OMCode, but commented each element out so that it would build, since this code isn’t yet available as a Nuget package.
